pkg/controller/envoy/ads_event.go

```diff
@@ -182,7 +182,7 @@ func (svc *ServiceEvent) handleCdsResponse(resp *service_discovery_v3.DiscoveryR
 	if !slices.EqualUnordered(svc.DynamicLoader.edsClusterNames, lastEdsClusterNames) {
 		svc.rqt = newAdsRequest(resource_v3.EndpointType, svc.DynamicLoader.edsClusterNames, svc.LastNonce.edsNonce)
 	} else {
-		svc.DynamicLoader.ClusterCache.Flush()
+		go svc.DynamicLoader.ClusterCache.Flush()
```

Comment:
If updating the ads map using asynchronous refresh, you need to remove the command in ut as follow.
```console
assert.Equal(t, sets.Set[string]{"ut-cluster": sets.Empty{}}, clusterNames)
```

Reply:
got